pegen/build.py

```diff
@@ -100,4 +91,41 @@ def build_parser(
     if compile_extension and output_file.endswith(".c"):
         compile_c_extension(output_file, verbose=verbose_c_extension)
 
+    return gen
+
+
+def build_parser_and_generator(
+    grammar_file,
+    output_file,
+    compile_extension=False,
+    verbose_tokenizer=False,
+    verbose_parser=False,
+    verbose_c_extension=False,
+):
+    """Generate rules, parser, tokenizer, parser generator for a given grammar
+
+    Args:
+        grammar_file (string): Path for the grammar file
+        output_file (string): Path for the output file
+        compile_extension (bool, optional): Whether to compile the C extension.
+          Defaults to False.
+        verbose_tokenizer (bool, optional): Whether to display additional output
+          when generating the tokenizer. Defaults to False.
+        verbose_parser (bool, optional): Whether to display additional output
+          when generating the parser. Defaults to False.
+        verbose_c_extension (bool, optional): Whether to display additional
+          output when compiling the C extension . Defaults to False.
+    """
+    parser, rules, tokenizer = build_parser(
+        grammar_file, verbose_tokenizer, verbose_parser
+    )
+    gen = build_generator(
+        tokenizer,
+        rules,
+        grammar_file,
+        output_file,
+        compile_extension,
+        verbose_c_extension,
+    )
+
     return rules, parser, tokenizer, gen
```

Comment:
Hm... It's kind of confusing that here we return `rules, parser, ...` whereas `build_parser()` returns `parser, rules, ...`. Can you make it so that here we just return an extra element but the first three are in the same order? (I honestly don't care which order, as long as it's consistent. :-)

Reply:
Ah yes, good catch. Done.